rust/connlib/tunnel/src/dns.rs

```diff
@@ -291,6 +291,11 @@ impl StubResolver {
 
                 vec![AllRecordData::Ptr(domain::rdata::Ptr::new(fqdn))]
             }
+            (Rtype::HTTPS, Some(_)) => {
+                anyhow::bail!(
+                    "Discarding HTTPS record query for resource {domain} because we can't mangle it"
```

Comment:
Instead of blackholing, should we return empty records? Or just a DNS server error? Or query not supported?

Reply:
Also, if we do that, it would be nice to test it with `tunnel_test`.

Reply:
Should we maybe answer _all_ other query types for resources with SERVFAIL? Forwarding them will never be correct, right? Or do we explicitly want to fetch things like TXT or MX records from the original resolver?